diff --git a/BUILD.gn b/BUILD.gn
index 05215d2..3f90fb6 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -11,15 +11,18 @@ source_set("header") {
 source_set("simdutf") {
   sources = [ "simdutf.cpp" ]
   public_deps = [ ":header" ]
-  cflags_cc = [
-    "-Wno-unused-const-variable",
-    "-Wno-unused-function",
-  ]

-  # TODO(crbug.com/483020951): Replace this with `-Wno-array-bounds-member` if
-  # that gets implemented. Then patch simdutf and remove this completely.
-  # Upstream issue: https://github.com/simdutf/simdutf/issues/938
-  cflags = [ "-Wno-array-bounds" ]
+  if (is_clang || !is_win) {
+    cflags_cc = [
+      "-Wno-unused-const-variable",
+      "-Wno-unused-function",
+    ]
+
+    # TODO(crbug.com/483020951): Replace this with `-Wno-array-bounds-member` if
+    # that gets implemented. Then patch simdutf and remove this completely.
+    # Upstream issue: https://github.com/simdutf/simdutf/issues/938
+    cflags = [ "-Wno-array-bounds" ]
+  }

   defines = []
   if (!allow_avx512) {
